src/main/java/com/cgm/infolab/controller/RoomApiController.java

```diff
@@ -0,0 +1,76 @@
+package com.cgm.infolab.controller;
+
+import com.cgm.infolab.db.model.ChatMessageEntity;
+import com.cgm.infolab.db.model.RoomEntity;
+import com.cgm.infolab.db.repository.ChatMessageRepository;
+import com.cgm.infolab.db.repository.RoomRepository;
+import com.cgm.infolab.model.RoomDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+public class RoomApiController {
+
+    private final RoomRepository roomRepository;
+    private final ChatMessageRepository chatMessageRepository;
+
+    private final Logger log = LoggerFactory.getLogger(RoomApiController.class);
+
+    @Autowired
+    public RoomApiController(RoomRepository roomRepository, ChatMessageRepository messageRepository) {
+        this.roomRepository = roomRepository;
+        this.chatMessageRepository = messageRepository;
+    }
+
+    // TODO: magari è necessario far si che le room prese siano soltanto quelle assegnate ad uno user,
```

Comment:
Questo sicuro. Tra i parametri di `getAllRooms` Spring ti può passare il `Principal`, che è lo user autenticato. Fare un match sullo username del principal è sufficiente, perchè il principal arriva dal sistema di autenticazione, che nel nostro caso è una Basic Authentication (username e password). Le room caricate da Db devono dipendere dallo user corrente.

La mia idea era di utilizzare il nome gli username dei due utenti che conversano (magari con un separatore tipo `-`), per nominare la stanza. Bisogna stabilire una regola per decidere quale nome utente mettere prima e quale dopo, essendo la stanza sempre la stessa vista da entrambi gli utenti. Non è banale, pensa alle query sul db. 
Prova :)

Reply:
Qui invece ho provato a usare il principal per controllare se ritornare una stanza o meno.
Diciamo che è un po' un casino.
Fino a che punto devo spingermi?
Nel senso che devo mettere il principal in ogni metodo di tutte le repository o basta che lo faccia per alcuni?
Inoltre provando ho già riscontrato un problema, cioè che il metodo getById della RoomRepository viene usato nel rowmapper della ChatMessageRepository, solo che quel metodo non lo chiamo io, quindi non posso ottenere il principal.

Reply:
Nei repository è sufficiente che passi solo lo `username`, anche come `String` nei soli metodi dove ti serve di fare il controllo. Ad esempio `public List<RoomEntity> getAfterDate(LocalDate dateLimit)` diventa `public List<RoomEntity> getByUserAfterDate(String username, LocalDate dateLimit)`

In un commento hai scritto qualcosa riguardo utilizzare le relazioni del Db sull'utente, anzichè una convenzione sui nomi della stanza. E' una soluzione decisamente migliore di quella che ti ho proposto io 😀. La relazione tra stanza e utenti c'è già, indirettamente, tramite i messaggi. I messaggi hanno una stanza e un sender (user)... quindi una stanza ha N users. O se ti crea trip mentale, puoi anche creare una nuova relazione diretta tra stanza e user, usando il liquibase... 
Una volta fatta la `JOIN` tra le tabelle, fai un `WHERE` sullo username.

Il flusso completo sarebbe: quando un utente inizia una nuova conversazione con un altro utente (dal FE usa la ricerca per cercare un utente, ci clicca e inizia la conversazione). [Storia per la creazione di una conversazione](https://github.com/Studiofarma/Infolab/issues/81). Quando clicca, invoca una API per la creazione della stanza. Quando apre una conversazione già esistente, invece chiama in `GET` il  controller che torna i messaggi di una stanza (`ChatApiMessageController`), come fa già adesso per i messaggi in General.

Non ho capito l'osservazione sul fatto che sia Spring a invocare il rowMapper. Può essere che confondi l'utilizzo della sintassi di una reference di funzione (`this:rowMapper`), senza parametri espliciti? Quello non è Spring, è solo sintassi Java per fare invocazione di una funzione come se fosse una lambda. Magari su questo possiamo fare un approfondimento in presenza.

Reply:
Riguardo all'ultima parte ho sbagliato a scrivere. Non è spring a chiamare il metodo, ma è il jdbcTemplate. Quello che intendevo è che non posso passargli a parametro il principal dall'esterno alla lambda.

Per il resto provo.

Reply:
Ho implementato la verifica controllando se il nome dell'utente compare nel nome della stanza assumendo la convenzione dei nomi delle stanze.
Non ho fatto quella delle relazioni perché ho avuto problemi con la gestione della relazione molti a molti (non l'ho mai fatto e ho letto che comporterebbe la creazione di una tabella dedicata) quindi prima di fare un disastro pensavo fosse meglio se ci confrontassimo un attimo su come fare.
Magari giovedì direttamente a questo punto.

Comunque per ora sembra funzionare.